src/xenia/cpu/xex_module.cc

```diff
@@ -41,11 +41,19 @@ XexModule::XexModule(Processor* processor, KernelState* kernel_state)
       processor_(processor),
       kernel_state_(kernel_state),
       xex_(nullptr),
+      xex_header_(nullptr),
       base_address_(0),
       low_address_(0),
       high_address_(0) {}
 
-XexModule::~XexModule() { xe_xex2_dealloc(xex_); }
+XexModule::~XexModule() {
+  xe_xex2_dealloc(xex_);
+
+  if (xex_header_) {
+    delete[] xex_header_;
```

Comment:
We should switch this to an std::vector so we don't need any of the cleanup code.
vector can be used like a dumb pointer with `.data()`, so things like:
`std::memcpy(xex_header_.data(), ...)` and such are possible. `xex_header_.resize(new_count);` will reallocate as needed.